module6/lessons/Week3/BinaryTrees.md

```diff
@@ -0,0 +1,119 @@
+---
+layout: page
+title: Binary Trees - CS Topic - Data Structure
+---
+
+## Introduction
+
+Today's lesson on the binary tree data structure comes to you in three parts. 
+
+**Part 1**: Independent research
+
+**Part 2**: Class walkthrough of key operations and time complexity
+
+**Part 3**: Diving into a C# implementation of a binary tree
+
+### Independent Learning
+Start by taking 30 minutes to independently research the binary tree Data Structure.
+
+I recommend that you start by reading these two articles:
+* [How To Not Be Stumped By Trees](https://medium.com/basecs/how-to-not-be-stumped-by-trees-5f36208f68a7)
+* [Leaf It Up to Binary Trees](https://medium.com/basecs/leaf-it-up-to-binary-trees-11001aaf746d)
+
+As you're researching, look for the answers to the following questions.
+
+1. What does it mean for a data structure to be non-linear?
+2. What is the difference between depth and height?
+3. How will inserting data into a binary tree differ from inserting data into a Linked List?
+
+## Walkthroughs
+### Binary Tree
+
+Let's walk through what it looks like to store data in a binary tree.  We will focus primarily on inserting data, and traversing the tree to find specific data points.
+
+<section class='instructor-notes' markdown='1'>
+
+This part should be pretty quick, we just want to illustrate inserting nodes into a tree using the next-left node.  You can use this [this random word generator](https://randomwordgenerator.com/) to pull data.
+
+We want students to see that a pure binary tree is not super useful because of how difficult it is to find data.  This should lead in to discussing a Binary Search Tree.
+
+</section>
+
+<section class='answer' markdown='1'>
+
+### Key Takeaways
```

Comment:
I like your Key Takeaways structure to help focus both students and instructors.